### DISCUSSION TITLE
Automatic Mathjax detection

### DISCUSSION THREAD

**Discussion by karuboniru** [walinejs/waline#377] [Q&A] [answered]
Could we make frontend to call `MathJax.typeset()` when Mathjax is already present in current page and disable katex there?

**Comment by lizheming** [comment]
We have support both mathjax and katex on the server side, view doc to get configuration https://waline.js.org/en/reference/server.html#markdown

**Reply by karuboniru** [reply]
How can I migrate LaTeX comments from Valine?

**Reply by Mister-Hope** [reply]
I am not sure how is Valine displaying math. What's actually in your database now? The raw tex or `<math>` or some other things?

**Reply by karuboniru** [reply]
> I am not sure how is Valine displaying math. What's actually in your database now? The raw tex or `<math>` or some other things?

It's raw in database, like `$...$`, which will be rendered by Mathjax when present.

**Reply by Mister-Hope** [reply]
Maybe we can add it in the migration tool later.

**Comment by Mister-Hope** [comment, marked as answer]
We support `texRenderer` option now.
